### PR TITLE
simplify osc sr code in submitting prj

### PR DESCRIPTION
**Pull request by lethliel**
Most of the stuff is now done in the API.
For just submitting all in one request the following is sufficient:

`<action type="submit"> <source project="%s" /> %s </action>`

We don't need package name or source package and targets anymore.
This is all handled by the API now.

For the `--seperate-requests` case the check for the entries in the linkinfo
is not working on most cases. So we handle this in do_submitrequest call
and skip the package if `X-Opensuse-Errorcode` is `missing_action` which means
basically nothing to do for this package

solves https://github.com/openSUSE/osc/issues/605

### CODE REVIEW

**Comment by lethliel** [comment]
@marcus-h: please review :)

**Comment by marcus-h** [comment]
On 2019-08-02 04:57:36 -0700, Marco Strigl wrote:
> diff --git a/osc/core.py b/osc/core.py
> index 032f62fa..04c675ca 100644
> --- a/osc/core.py
> +++ b/osc/core.py
> @@ -4268,6 +4268,8 @@ def create_submit_request(apiurl,
>              tproject = project.get('name')
>              r = create_maintenance_request(apiurl, src_project, [src_package], tproject, dst_project, src_update, message)
>              r = r.reqid
> +        elif e.hdrs.get('X-Opensuse-Errorcode') == "missing_action":
> +            r = e.hdrs.get('X-Opensuse-Errorcode')

Hmm this would change the api and may break 3rd party code. What about
moving this check into do_submitrequest? Alternatively, we could simply
error out (as we do now).

**Comment by lethliel** [comment]
On 8/15/19 5:54 PM, Marcus Hüwe  wrote:
> On 2019-08-02 04:57:36 -0700, Marco Strigl wrote:
>> diff --git a/osc/core.py b/osc/core.py
>> index 032f62fa..04c675ca 100644
>> --- a/osc/core.py
>> +++ b/osc/core.py
>> @@ -4268,6 +4268,8 @@ def create_submit_request(apiurl,
>> tproject = project.get('name')
>> r = create_maintenance_request(apiurl, src_project, [src_package], tproject, dst_project, src_update, message)
>> r = r.reqid
>> + elif e.hdrs.get('X-Opensuse-Errorcode') == "missing_action":
>> + r = e.hdrs.get('X-Opensuse-Errorcode')
> 
> Hmm this would change the api and may break 3rd party code. What about
> moving this check into do_submitrequest? Alternatively, we could simply
> error out (as we do now).
> 

Ok. I don't see any scenarios where people would rely on raising on X-Opensuse-Errorcode is missing_action,
but you are probably right that the caller should handle this. 

Moved the exception handling for missing_action to do_submitrequest and edited the PR and commit description.

**Comment by marcus-h** [comment]
On 2019-08-16 05:56:15 -0700, Marco Strigl wrote:
> On 8/15/19 5:54 PM, Marcus Hüwe  wrote:
> > On 2019-08-02 04:57:36 -0700, Marco Strigl wrote:
> >> diff --git a/osc/core.py b/osc/core.py
> >> index 032f62fa..04c675ca 100644
> >> --- a/osc/core.py
> >> +++ b/osc/core.py
> >> @@ -4268,6 +4268,8 @@ def create_submit_request(apiurl,
> >> tproject = project.get('name')
> >> r = create_maintenance_request(apiurl, src_project, [src_package], tproject, dst_project, src_update, message)
> >> r = r.reqid
> >> + elif e.hdrs.get('X-Opensuse-Errorcode') == "missing_action":
> >> + r = e.hdrs.get('X-Opensuse-Errorcode')
> >
> > Hmm this would change the api and may break 3rd party code. What about
> > moving this check into do_submitrequest? Alternatively, we could simply
> > error out (as we do now).
> >
>
> Ok. I don't see any scenarios where people would rely on raising on X-Opensuse-Errorcode is missing_action,
> but you are probably right that the caller should handle this.
>
Let's consider the following made up code-snippet:

...
rid = create_submit_request(apiurl, src_prj, src_pkg, tgt_prj)
if rid is None:
    # would indicate a broken xml, but well...
    sys.exit('failure')
# ok SR was created, insert rid in some db whatever...
...

Now, in case of a missing_action, the code above would continue and
could fail in some very strange way...

Back to your patch: "cd prj; osc sr <target_prj>" will ignore the specified
target project. Otherwise, the patch seems to be OK from a refactoring POV!
(Some parts of the old code are pretty buggy/broken - we should probably
tackle them in another patch.)
